Fix 404 thumbnails and posting in Wired-7 site

Wired-7 opened the pull request:
Recently I started using .webp in Wired-7 thumbnails. By default Kuroba expects them to be .png as vichan doesn't serve the thumbnail extension in the API. In consequence, all new thumbnails were 404.

Right now we have mixed thumb extensions: old posts have png and new ones have webp. Until all posts get recycled I added a new field in our API with the thumbnail name ("th"). With this merge I want to implement a new Wired7Api and Wired7Endpoints to use this field. Those classes are almost the same but using the new field and updating a few imports in order to clean code.

I also update "makeArgument" from "SiteEndpoints.java" as it was implemented multiple times with fixed arguments.

Wired-7 committed:
Create a package for Wired-7 site

K1rakishou commented on Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/sites/wired7/Wired7Api.kt:
```diff
@@ -0,0 +1,351 @@
+
+    val site = siteManager.bySiteDescriptor(chanReaderProcessor.chanDescriptor.siteDescriptor())
+      ?: return
+    val board = boardManager.byBoardDescriptor(chanReaderProcessor.chanDescriptor.boardDescriptor())
```
You can probably do the same thing as here https://github.com/K1rakishou/Kuroba-Experimental/blob/b981bc96bf7b94653cda1dc1191817d12e5d5bfa/Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/common/FutabaChanReader.kt#L103 which will allow users opening site boards without even loading them first.

Wired-7 replied:
Fixed in [b4bfc0c](https://github.com/K1rakishou/Kuroba-Experimental/pull/750/commits/b4bfc0ca420ca71230f00819c457aa6c6b177461)

K1rakishou commented on Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/sites/wired7/Wired7Api.kt:
```diff
@@ -0,0 +1,351 @@
+  }
+
+  @Throws(IOException::class)
+  private fun readPostImage(
```
I think you don't need this since this is a workaround for 8kun bug.

K1rakishou commented on Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/sites/wired7/Wired7Api.kt:
```diff
@@ -0,0 +1,351 @@
+    board: ChanBoard,
+    endpoints: SiteEndpoints
+  ): ChanPostImage? {
+    try {
```
I think you don't need this try/catch block since this is a workaround for 8kun bug.

Wired-7 replied:
Fixed in [b4bfc0c](https://github.com/K1rakishou/Kuroba-Experimental/pull/750/commits/b4bfc0ca420ca71230f00819c457aa6c6b177461)

Wired-7 commented:
Thanks for the review I will update the pull when I have a bit of spare time.

K1rakishou committed:
Some minor renamings + update fsaf to the latest version.

Wired-7 committed:
Merge branch 'K1rakishou:develop' into develop

Wired-7 committed:
Update LainchanActions and Lainchan

Wired-7 commented:
I don't know what changed between Kuroba and Kuroba Experimental but now it doesn't detect how many files we can post in vichan. Looks like now we have to configure it with "setPostingLimitationInfo", I didn't know this.

- In Wired-7: I updated Wired-7 with the pull request comments and to allow our 3 file limit and 20MB size per post. 
- LainchanActions: I wanted to reuse LainchanActions in Wired-7. LainchanActions only checked one file per post (using firstFileOrNull when getting the files). After my change now it includes all files. It doesn't control how many files are added as it now is controlled by "setPostingLimitationInfo".
- Lainchan: In order to modify LainchanActions I also updated Lainchan from Java to Koltin. It also had the same problem as Wired-7 that users could only post one file per post.

I still don't want to merge because something is adding a "spoiler" into the FormDataParts in Lainchan. Vichan doesn't check if "spoiler" is true or false, it only checks if it exists, so all files uploaded to lainchan are spoiled. I have to check if this already happend before my change, but I don't see where I could have added that...

K1rakishou committed:
Merge remote-tracking branch 'origin/develop' into develop

K1rakishou committed:
Merge branch 'develop' into Wired-7_develop

# Conflicts:
#	Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/SiteRegistry.kt

K1rakishou commented:
>Vichan doesn't check if "spoiler" is true or false, it only checks if it exists, so all files uploaded to lainchan are spoiled.

I just checked it on the site and apparently you can't set spoiler flag for individual files so it's either all files are marked as spoiler or none. Earlier you could only attach one file so it was not a problem.

If it's possible on your site then it should be handled separately for your site.

K1rakishou committed:
Resolve merge conflicts, set spoiler flag to true if any of the attachments marked with spoiler.

K1rakishou committed:
Rename the parameter so it's obvious that it's supposed to be in bytes.

K1rakishou commented:
Also spoilered media doesn't work on Lainchan and there is nothing in the json to indicate whether a media is spoilered or not.

K1rakishou committed:
These should probably be megabytes not kilobytes.

K1rakishou commented:
I also changed the parameter of ConstantMaxTotalSizeInfo for Wired and Lainchan because they were using incorrect maxSize parameters (kilobytes instead of megabytes).

K1rakishou commented:
Other than that the rest seems fine.

Wired-7 commented:
Thanks for the corrections!

> Also spoilered media doesn't work on Lainchan and there is nothing in the json to indicate whether a media is spoilered or not.

In vichan you can't spoiler files individually, there is only one checkbox that affects everything.

I could reproduce the spoiler bug I mentioned with the official develop so it isn't a problem related to this pull changes. I'd like to merge this as it is not related to the spoiler bug, it should be added as a new issue. That bug is probably caused by LainchanAntispam including "spolier" field among the rest of the fields while it should only be included when its checked by the user.
